src/leaflike/bookmarks.clj

```diff
@@ -166,16 +168,21 @@
 (def search-bookmarks-view (partial bookmarks-list :search-bookmarks))
 
 (defn create-view
-  [request]
+  [{:keys [params] :as request}]
   (let [username (get-in request [:session :username])
         error-msg (get-in request [:flash :error-msg])
         user      (hutils/get-user request)
+        next-url (:next params "/bookmarks")
+        prefill-url (:url params)
+        prefill-title (:title params)
         all-tags (map :name (tags-db/fetch-tags {:user-id (:id user)}))]
     (-> (res/response (layout/user-view "Add Bookmark"
                                         username
                                         (views/bookmark-form anti-forgery/*anti-forgery-token*
-                                                             "/bookmarks/add"
-                                                             {:all-tags all-tags})
+                                                             (str "/bookmarks/add?next=" next-url)
+                                                             {:all-tags all-tags
+                                                              :url prefill-url
```

Comment:
The idea we're trying to get across here is that `:url` and `:title` may be present in the incoming params and if they are they get carried forward. So we don't need to have the intermediate `prefill-` binding, that would be redundant. Ideally you can just pass down the params and assoc `all-tags` so that the downstream fn can destructure as it sees fit.